lib/services/s3.js

```diff
@@ -162,10 +162,20 @@ AWS.util.update(AWS.S3.prototype, {
       return;
     }
     var bucket = req.params && req.params.Bucket;
-    if (bucket && bucket.indexOf('/') >= 0) {
-      var msg = 'Bucket names cannot contain forward slashes. Bucket: ' + bucket;
-      throw AWS.util.error(new Error(),
-        { code: 'InvalidBucket', message: msg });
+    var key = req.params && req.params.Key;
+    var slashIndex = bucket && bucket.indexOf('/');
+    if (bucket && slashIndex >= 0) {
+      if (typeof key === 'string') {
+        req.params = AWS.util.copy(req.params);
+        // Need to include trailing slash to match sigv2 behavior
+        var prefix = bucket.substr(slashIndex + 1) || '';
+        req.params.Key = prefix + '/' + key;
+        req.params.Bucket = bucket.substr(0, slashIndex);
+      } else {
+        var msg = 'Bucket names cannot contain forward slashes. Bucket: ' + bucket;
```

Comment:
If a user did not supply a key at all, then they would receive a missing required param error. That said, it is possible to turn off param validation so maybe I should just make sure Key exists as a possible parameter.